app/controllers/users_controller.rb

```diff
@@ -8,21 +8,60 @@ def show
   end
 
   def update
-    if @user.update(user_params)
-      redirect_to @user, notice: 'The user has been updated correctly.'
+    pass_params = password_params.to_h
+    exist_pass_arg = pass_params.any? { |k, v| !v.empty? and !v.nil? }
```

Comment:
I think devise provides two methods to update the user with and without password

``user. update_with_password || user.update_without_password`

also it would be cleaner if you only delete the password fields if they are empty

```ruby
   params[:user].delete(:password) if params[:user][:password].blank?
    params[:user].delete(:password_confirmation) if params[:user][:password_confirmation].blank?
```
The lines above can be within a before_action method
and then you can add a condition like so:
```ruby
 if user_params.password.present?
     user.update_with_password(user_params)
else
  user.update_without_password(user_params)
end
```

Reply:
- I don't use update_with_password method of devise due to it doesn't match the fields password
   and password_confirmation. http://www.rubydoc.info/github/plataformatec/devise/Devise%2FModels%2FDatabaseAuthenticatable%3Aupdate_with_password